test/query.test.js

```diff
@@ -2708,17 +2710,86 @@ describe('query', () => {
     }
   })
 
-  // TODO Create tests once work is done in Core
-  test.skip('current_identity', () => {})
+  // API v4 AccessProvider functions
+  // TODO: Add tests for JWTs using Auth0/Okta
+
+  test('current_identity returns object when authed', async () => {
+    // Create a new Collection
+    const newCollection = await client.query(
+      query.Create(collectionRef, { credentials: { password: 'sekrit' } })
+    )
+
+    // Login with permissions for newly created Collection
+    const newLogin = await client.query(
+      query.Login(newCollection['ref'], { password: 'sekrit' })
+    )
+
+    // Get the newly scoped client
+    const newClient = await util.getClient({ secret: newLogin['secret'] })
+
+    // Retrieve the current identity
+    const currentIdentity = await newClient.query(query.CurrentIdentity())
+    expect(currentIdentity).toEqual(newCollection.ref)
+    expect(currentIdentity).toEqual(newLogin.instance)
+    expect(currentIdentity).toBeInstanceOf(Object)
+  })
+
+  test('current_identity fails when no identity present', async () => {
+    try {
+      await adminClient.query(query.CurrentIdentity())
+    } catch (err) {
+      expect(err).toBeInstanceOf(errors.BadRequest)
+    }
+  })
 
-  // TODO Add test once Core work has been done
-  test.skip('has_current_identity', () => {})
+  test('has_current_identity returns true when identity present', async () => {
+    // Create a new Collection
+    const newCollection = await client.query(
+      query.Create(collectionRef, { credentials: { password: 'sekrit' } })
+    )
 
-  // TODO Finish test after Core work is done
-  test.skip('current_token', () => {})
+    // Login with permissions for newly created Collection
+    const newLogin = await client.query(
+      query.Login(newCollection['ref'], { password: 'sekrit' })
+    )
+
+    // Get the newly scoped client
+    const newClient = await util.getClient({ secret: newLogin['secret'] })
+
+    // Retrieve the current identity
+    const currentIdentity = await newClient.query(query.CurrentIdentity())
+    expect(currentIdentity).toBeTruthy()
+  })
+
+  test('has_current_identity returns false without identity present', async () => {
+    const currentIdentity = await adminClient.query(query.HasCurrentIdentity())
+    expect(currentIdentity).toBeFalsy()
+  })
+
+  test('current_token returns proper token', async () => {
+    const currentToken = await adminClient.query(query.CurrentToken())
+    expect(currentToken).toEqual(adminKey.ref)
+  })
+
+  test('current_token fails when unauthenticated', async () => {
+    try {
+      await adminClient.query(query.CurrentToken())
```

Comment:
I have reasons to believe this call is not throwing an exception, then, the `catch` block is not executed, making the test pass, but shouldn't, on try/catch tests like this I usually do this pattern
```
try {
  callTheTestingFunction();

  fail(); //if the testing function didn't throw an error, it will fail anyway
} catch(err) {
  assert on err
}
```

Reply:
@marrony Good catch! The query was not failing with the `adminClient` and thus not jumping into the `catch` block. I've revised the test to use the normal `client` which successfully throws so I can make the proper assertion in the `catch` 👍